From c82fd7688990501771033551481add87fd724bd2 Mon Sep 17 00:00:00 2001
From: Harshesh Valera <harshesh.v@variscite.com>
Date: Tue, 24 Apr 2018 11:17:49 -0700
Subject: [PATCH 1/2] Ported SPL-Uboot patch from 7.1.2

Signed-off-by: Harshesh Valera <harshesh.v@variscite.com>
---
 core/Makefile | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/core/Makefile b/core/Makefile
index a650565..f3bec8f 100644
--- a/core/Makefile
+++ b/core/Makefile
@@ -1353,7 +1353,10 @@ $(TARGET_BOOTLOADER_IMAGE): $(BOOTLOADER_OUT)
 		$(MAKE) -C $(UBOOT_IMX_PATH)/uboot-imx/ O=$(realpath $(BOOTLOADER_OUT)) $$UBOOT_CONFIG $(BOOTLOADER_ENV); \
 		$(MAKE) -C $(UBOOT_IMX_PATH)/uboot-imx/ mrproper $(BOOTLOADER_ENV); \
 		$(MAKE) -C $(UBOOT_IMX_PATH)/uboot-imx/ O=$(realpath $(BOOTLOADER_OUT)) $(BOOTLOADER_ENV) || exit 1; \
-		install -D $(BOOTLOADER_OUT)/u-boot$(TARGET_DTB_POSTFIX).$(TARGET_BOOTLOADER_POSTFIX) $(PRODUCT_OUT)/u-boot-$$UBOOT_PLATFORM.imx; \
+		install -D $(BOOTLOADER_OUT)/u-boot$(TARGET_DTB_POSTFIX).$(TARGET_BOOTLOADER_POSTFIX) $(PRODUCT_OUT)/u-boot-$$UBOOT_PLATFORM.$(TARGET_BOOTLOADER_POSTFIX); \
+		if [ "$(TARGET_BOOTLOADER_POSTFIX)" == "img" ]; then \
+			install -D $(BOOTLOADER_OUT)/SPL $(PRODUCT_OUT)/SPL-$$UBOOT_PLATFORM; \
+		fi; \
 		if [ $(UBOOT_POST_PROCESS) = true ]; then \
 		echo "build post process" ; \
 		$(call build_uboot, $(TARGET_BOOTLOADER_POSTFIX), $$UBOOT_PLATFORM) \
-- 
2.7.4

