diff --git a/device/fsl/common/wifi/Android.mk b/device/fsl/common/wifi/Android.mk
index 98a381d..28f83d8 100644
--- a/device/fsl/common/wifi/Android.mk
+++ b/device/fsl/common/wifi/Android.mk
@@ -24,6 +24,7 @@
 # components.
 
 ifeq ($(TARGET_BOARD_PLATFORM),$(filter $(TARGET_BOARD_PLATFORM),imx6 imx7))
+ifneq ($(USES_TI_MAC80211),true)
 
 LOCAL_PATH := $(call my-dir)
 
@@ -52,5 +53,6 @@ LOCAL_MODULE_PATH  := $(TARGET_OUT_ETC)/wifi
 include $(BUILD_PREBUILT)
 
 endif
+endif
 
 include $(call first-makefiles-under,$(LOCAL_PATH))
