diff --git a/Android.mk b/Android.mk
index 090f6c0..bb44211 100644
--- a/Android.mk
+++ b/Android.mk
@@ -4,7 +4,7 @@ include $(CLEAR_VARS)
  
 LOCAL_MODULE_TAGS := eng
 LOCAL_C_INCLUDES += $(LOCAL_PATH) $(LOCAL_PATH)/include
-LOCAL_SRC_FILES := tools/i2cbusses.c tools/util.c
+LOCAL_SRC_FILES := tools/i2cbusses.c tools/util.c lib/smbus.c
 LOCAL_MODULE := i2c-tools
 include $(BUILD_STATIC_LIBRARY)
  
