From 803af1f428ee3593f13cf5b2c2437ce1784119a6 Mon Sep 17 00:00:00 2001
From: varigigi <pierluigi.p@variscite.com>
Date: Sat, 27 May 2017 15:28:59 +0200
Subject: [PATCH] add utilities to manage kernel 'machine model'

---
 init/util.cpp | 34 ++++++++++++++++++++++++++++++++++
 init/util.h   |  2 ++
 2 files changed, 36 insertions(+)

diff --git a/init/util.cpp b/init/util.cpp
index ef82c33..7b0dc77 100644
--- a/init/util.cpp
+++ b/init/util.cpp
@@ -458,6 +458,40 @@ void get_soc_name(char *soc)
      fclose(fp);
 }
 
+void get_machine_model(char *machine, int size)
+{
+    const char *machinemodel = "/sys/devices/soc0/machine";
+    FILE *fp;
+
+    fp = fopen(machinemodel, "r");
+    if (fp == NULL) {
+        ERROR("Failed reading %s\n", machinemodel);
+        return;
+    }
+
+    if (fgets(machine, size, fp) == NULL) {
+        ERROR("fgets fail\n");
+        fclose(fp);
+        return;
+     }
+
+     fclose(fp);
+}
+
+int match_machine_model(const char *machine)
+{
+    char machinemodel[80];
+    char *pch;
+
+    get_machine_model(machinemodel, sizeof(machinemodel));
+
+    pch = strstr(machinemodel, machine);
+    if (pch == NULL) {
+        NOTICE("'%s' doesn't match machine model '%s'\n", machine, machinemodel);
+        return 0;
+    }
+    return 1;
+}
 
 int make_dir(const char *path, mode_t mode)
 {
diff --git a/init/util.h b/init/util.h
index f6710ad..ce6d27b 100644
--- a/init/util.h
+++ b/init/util.h
@@ -61,6 +61,8 @@ void open_devnull_stdio(void);
 void import_kernel_cmdline(bool in_qemu,
                            std::function<void(const std::string&, const std::string&, bool)>);
 void get_soc_name(char *soc);
+void get_machine_model(char *machine, int size);
+int match_machine_model(const char *machine);
 int make_dir(const char *path, mode_t mode);
 int restorecon(const char *pathname);
 int restorecon_recursive(const char *pathname);
-- 
2.7.4

