From a936ca9810df907ed34476ee4a312061827e0680 Mon Sep 17 00:00:00 2001
From: mohit <mohit.p@variscite.com>
Date: Wed, 11 May 2022 17:10:30 -0700
Subject: [PATCH] imx8mm_var_som: Allow parsing additional board specific 
 permissions.

---
 init/ueventd.cpp | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/init/ueventd.cpp b/init/ueventd.cpp
index 331255b1e..fad1a62b6 100644
--- a/init/ueventd.cpp
+++ b/init/ueventd.cpp
@@ -271,8 +271,9 @@ static UeventdConfiguration GetConfiguration() {
     // TODO: Remove these legacy paths once Android S is no longer supported.
     if (android::base::GetIntProperty("ro.product.first_api_level", 10000) <= __ANDROID_API_S__) {
         auto hardware = android::base::GetProperty("ro.hardware", "");
+	auto board = android::base::GetProperty("ro.boot.board", "");
         return ParseConfig({"/system/etc/ueventd.rc", "/vendor/ueventd.rc", "/odm/ueventd.rc",
-                            "/ueventd." + hardware + ".rc"});
+			"/odm/ueventd.rc", "/ueventd." + hardware + ".rc", "/vendor/ueventd."+ board + ".rc"});
     }
 
     return ParseConfig({"/system/etc/ueventd.rc"});
-- 
2.36.1

