From 0305a2dfa9f0833413b8facd14c07f3bba65da0c Mon Sep 17 00:00:00 2001
From: varigigi <pierluigi.p@variscite.com>
Date: Mon, 1 May 2017 17:33:43 +0200
Subject: [PATCH 2/3] allow skipping write on console

---
 init/init.cpp | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/init/init.cpp b/init/init.cpp
index f215ae9..bcba4bb 100644
--- a/init/init.cpp
+++ b/init/init.cpp
@@ -364,6 +364,10 @@ static int console_init_action(const std::vector<std::string>& args)
         have_console = 1;
     close(fd);
 
+    if (property_get("ro.boot.skip_write_on_console") == "1") {
+        return 0;
+    }
+
     fd = open("/dev/tty0", O_WRONLY | O_CLOEXEC);
     if (fd >= 0) {
         const char *msg;
-- 
2.7.4

