From daa05d8bce005756118b61bf3cffe2be30965032 Mon Sep 17 00:00:00 2001
From: varigigi <pierluigi.p@variscite.com>
Date: Sat, 27 May 2017 15:42:24 +0200
Subject: [PATCH] for VAR-DART, use dedicated fstab

---
 init/builtins.cpp | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/init/builtins.cpp b/init/builtins.cpp
index 44217f0..ff02b81 100644
--- a/init/builtins.cpp
+++ b/init/builtins.cpp
@@ -604,6 +604,7 @@ static int do_mount_all(const std::vector<std::string>& args) {
     bool queue_event = true;
     int mount_mode = MOUNT_MODE_DEFAULT;
     const char* fstabfile = args[1].c_str();
+    const char *fstabdart = "/fstab_dart.freescale";
     std::size_t path_arg_end = args.size();
 
     for (na = args.size() - 1; na > 1; --na) {
@@ -618,7 +619,7 @@ static int do_mount_all(const std::vector<std::string>& args) {
         }
     }
 
-    int ret =  mount_fstab(fstabfile, mount_mode);
+    int ret = mount_fstab(match_machine_model("VAR-DART") ? fstabdart : fstabfile, mount_mode);
 
     if (import_rc) {
         /* Paths of .rc files are specified at the 2nd argument and beyond */
-- 
2.7.4

