From 685ad393be0ab528f637ad734b2bb576672a2d14 Mon Sep 17 00:00:00 2001
From: varigigi <pierluigi.p@variscite.com>
Date: Wed, 19 Jul 2017 20:21:06 +0200
Subject: [PATCH] when /etc/recovery.fstab missing, fallback to
 fstab.${ro.hardware}

---
 roots.cpp | 13 ++++++++++++-
 1 file changed, 12 insertions(+), 1 deletion(-)

diff --git a/roots.cpp b/roots.cpp
index 4b8cd12..aa73212 100644
--- a/roots.cpp
+++ b/roots.cpp
@@ -52,7 +52,18 @@ void load_volume_table()
     fstab = fs_mgr_read_fstab("/etc/recovery.fstab");
     if (!fstab) {
         LOGE("failed to read /etc/recovery.fstab\n");
-        return;
+
+        char fstab_path[PATH_MAX+1] = "/fstab.";
+        if (!property_get("ro.hardware", fstab_path+strlen(fstab_path), "")) {
+            LOGE("failed to get ro.hardware");
+            return;
+        }
+
+        fstab = fs_mgr_read_fstab(fstab_path);
+        if (!fstab) {
+            LOGE("failed to read %s", fstab_path);
+            return;
+        }
     }
 
     ret = fs_mgr_add_entry(fstab, "/tmp", "ramdisk", "ramdisk");
-- 
2.7.4

