From 3d9dbc9a6463a846cb24660a217ef2bcedbc87a1 Mon Sep 17 00:00:00 2001
From: varigigi <pierluigi.p@variscite.com>
Date: Fri, 7 Apr 2017 22:45:14 +0200
Subject: [PATCH] allow disabling automatic build of fsl wifi

---
 common/wifi/Android.mk | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/common/wifi/Android.mk b/common/wifi/Android.mk
index 98a381d..28f83d8 100644
--- a/common/wifi/Android.mk
+++ b/common/wifi/Android.mk
@@ -24,6 +24,7 @@
 # components.
 
 ifeq ($(TARGET_BOARD_PLATFORM),$(filter $(TARGET_BOARD_PLATFORM),imx6 imx7))
+ifneq ($(USES_TI_MAC80211),true)
 
 LOCAL_PATH := $(call my-dir)
 
@@ -52,5 +53,6 @@ LOCAL_MODULE_PATH  := $(TARGET_OUT_ETC)/wifi
 include $(BUILD_PREBUILT)
 
 endif
+endif
 
 include $(call first-makefiles-under,$(LOCAL_PATH))
-- 
2.7.4

