From a442c3a5f8ae85b0b63deb8a2803f79f72ae0c11 Mon Sep 17 00:00:00 2001
From: mohit <mohit.p@variscite.com>
Date: Thu, 21 Jan 2021 16:18:59 -0800
Subject: [PATCH] Android: Changes required to compile SCFW

---
 common/tools/imx-make.sh | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/common/tools/imx-make.sh b/common/tools/imx-make.sh
index ee7b83f4..a20ed96a 100755
--- a/common/tools/imx-make.sh
+++ b/common/tools/imx-make.sh
@@ -105,6 +105,10 @@ for arg in ${args[*]} ; do
     esac
 done
 
+if [ ${TARGET_PRODUCT} = "som_mx8qx" ]; then
+	source $ANDROID_BUILD_TOP/device/variscite/imx8q/som_mx8q/uboot-firmware/scfw_build.sh ${TARGET_PRODUCT}
+fi
+
 # if bootloader and kernel not in arguments, all need to be made
 if [ ${build_bootloader_kernel_flag} -eq 0 ] && [ ${build_android_flag} -eq 0 ]; then
     build_bootloader="bootloader";
-- 
2.25.1

