From e16cf0d5a9b9651770b5511ab990baa0fa321451 Mon Sep 17 00:00:00 2001
From: varigigi <pierluigi.p@variscite.com>
Date: Fri, 7 Apr 2017 22:46:58 +0200
Subject: [PATCH] allow recovery on boards without 3 buttons

---
 common/recovery/recovery_ui.cpp | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/common/recovery/recovery_ui.cpp b/common/recovery/recovery_ui.cpp
index 0a85307..233c77a 100644
--- a/common/recovery/recovery_ui.cpp
+++ b/common/recovery/recovery_ui.cpp
@@ -38,10 +38,14 @@ const char* ITEMS[] = { "reboot system now",
 class ImxUI : public ScreenRecoveryUI {
   public:
     virtual KeyAction CheckKey(int key) {
+      if (HasThreeButtons()) {
         if (IsKeyPressed(KEY_POWER) && key == KEY_VOLUMEUP) {
             return TOGGLE;
         }
         return ENQUEUE;
+      } else {
+        return IGNORE;
+      }
     }
 };
 
@@ -63,8 +67,14 @@ class ImxDevice : public Device {
               case KEY_VOLUMEUP:
                 return kHighlightUp;
 
+              case KEY_ENTER:
               case KEY_POWER:
                 return kInvokeItem;
+
+              default:
+                // If you have all of the above buttons, any other buttons
+                // are ignored. Otherwise, any button cycles the highlight.
+                return GetUI()->HasThreeButtons() ? kNoAction : kHighlightDown;
             }
         }
 
-- 
2.7.4

