From c25cd97eed35c6ab45e0d595c7ce3f547e515c52 Mon Sep 17 00:00:00 2001
From: Mohit Singh <pal.mohit.singh@gmail.com>
Date: Mon, 2 Mar 2020 16:08:53 -0800
Subject: [PATCH] imx8mm-var-som: Add support for SOM revision 1.1

---
 iMX8M/soc.mak | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/iMX8M/soc.mak b/iMX8M/soc.mak
index f0acef8..b01b83a 100644
--- a/iMX8M/soc.mak
+++ b/iMX8M/soc.mak
@@ -137,7 +137,7 @@ u-boot-ddr4-evk.itb: $(dtbs_ddr4_evk)
 	TEE_LOAD_ADDR=$(TEE_LOAD_ADDR) ATF_LOAD_ADDR=$(ATF_LOAD_ADDR) ./mkimage_fit_atf.sh $(dtbs_ddr4_evk) > u-boot-ddr4-evk.its
 	./mkimage_uboot -E -p 0x3000 -f u-boot-ddr4-evk.its u-boot-ddr4-evk.itb
 
-dtbs_lpddr4_ddr4_evk = fsl-$(PLAT)-var-dart.dtb fsl-$(PLAT)-var-som.dtb
+dtbs_lpddr4_ddr4_evk = fsl-$(PLAT)-var-dart.dtb fsl-$(PLAT)-var-som.dtb fsl-$(PLAT)-var-som-rev10.dtb
 u-boot-lpddr4-ddr4-evk.itb: $(dtbs_lpddr4_ddr4_evk)
 	./$(PAD_IMAGE) bl31.bin
 	TEE_LOAD_ADDR=$(TEE_LOAD_ADDR) ATF_LOAD_ADDR=$(ATF_LOAD_ADDR) ./mkimage_fit_atf.sh $(dtbs_lpddr4_ddr4_evk) > u-boot-lpddr4-ddr4-evk.its
-- 
2.25.0

