From 35e84f3506191761165353b34e84391526e78052 Mon Sep 17 00:00:00 2001
From: Mohit Singh <mohit.p@variscite.com>
Date: Wed, 30 Sep 2020 11:01:24 -0700
Subject: [PATCH 2/2] soc.mak: Add support for imx8mq_var_dart

Signed-off-by: Harshesh Valera <harshesh.v@variscite.com>
---
 iMX8M/soc.mak | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/iMX8M/soc.mak b/iMX8M/soc.mak
index 5a42555..3cb1e45 100644
--- a/iMX8M/soc.mak
+++ b/iMX8M/soc.mak
@@ -133,9 +133,9 @@ u-boot-atf-tee.bin: u-boot.bin bl31.bin tee.bin
 clean:
 	@rm -f $(MKIMG) u-boot-atf.bin u-boot-atf-tee.bin u-boot-spl-ddr.bin u-boot.itb u-boot.its u-boot-ddr3l.itb u-boot-ddr3l.its u-boot-spl-ddr3l.bin u-boot-ddr4.itb u-boot-ddr4.its u-boot-spl-ddr4.bin u-boot-ddr4-evk.itb u-boot-ivt.itb u-boot-ddr4-evk.its $(OUTIMG)
 
-dtbs = evk.dtb
+dtbs = $(PLAT)-var-dart.dtb $(PLAT)-var-som.dtb
 $(dtbs):
-	./$(DTB_PREPROC) $(PLAT)-evk.dtb $(dtbs)
+	./$(DTB_PREPROC) $(dtbs)
 
 u-boot.itb: $(dtbs)
 	./$(PAD_IMAGE) tee.bin
-- 
2.25.1

