From 1d916a4a1f2bfc88d6d487a21e6e3b5ac3f35712 Mon Sep 17 00:00:00 2001
From: Harshesh Valera <harshesh.v@variscite.com>
Date: Fri, 25 Feb 2022 17:08:56 -0800
Subject: [PATCH] imx8mp_var_dart: Fix preview fps for imx8mp platforms

Signed-off-by: Harshesh Valera <harshesh.v@variscite.com>
---
 src/com/android/camera/VideoModule.java | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/src/com/android/camera/VideoModule.java b/src/com/android/camera/VideoModule.java
index 7b996147f..7aa879064 100644
--- a/src/com/android/camera/VideoModule.java
+++ b/src/com/android/camera/VideoModule.java
@@ -1660,6 +1660,11 @@ public class VideoModule extends CameraModule
             mCameraSettings.setPreviewFrameRate(mProfile.videoFrameRate);
         }
 
+	if (Build.PRODUCT.contains("dart_mx8mp")) {
+		mProfile.videoFrameRate = 30;
+		mCameraSettings.setPreviewFrameRate(30);
+	}
+
         enableTorchMode(Keys.isCameraBackFacing(settingsManager, mAppController.getModuleScope()));
 
         // Set zoom.
-- 
2.25.1

