From 2db2dcef88bfefbf9c62dfbe7fa1ffed70e4777a Mon Sep 17 00:00:00 2001
From: varigigi <pierluigi.p@variscite.com>
Date: Wed, 17 Feb 2016 14:55:33 +0100
Subject: [PATCH 21/34] display timing: porting support for alternative display

---
 drivers/video/of_display_timing.c | 41 ++++++++++++++++++++++++++++++++-------
 1 file changed, 34 insertions(+), 7 deletions(-)

diff --git a/drivers/video/of_display_timing.c b/drivers/video/of_display_timing.c
index ba5b40f..6e71c69 100644
--- a/drivers/video/of_display_timing.c
+++ b/drivers/video/of_display_timing.c
@@ -13,6 +13,19 @@
 #include <video/display_timing.h>
 #include <video/of_display_timing.h>
 
+static int screen_alternate;
+static int __init alternate_screen_type(char *options)
+{
+
+	if (strcmp(options, "yes"))
+		screen_alternate= 0;
+	else
+		screen_alternate = 1;
+
+	return 0;
+}
+__setup("screen_alternate=", alternate_screen_type);
+
 /**
  * parse_timing_property - parse timing_entry from device_node
  * @np: device_node with the property
@@ -147,11 +160,18 @@ struct display_timings *of_get_display_timings(struct device_node *np)
 		return NULL;
 	}
 
-	timings_np = of_get_child_by_name(np, "display-timings");
+	if (screen_alternate == 1)
+		timings_np = of_find_node_by_name(np, "display-timings-alternate");
+	else
+		timings_np = NULL;
+
 	if (!timings_np) {
-		pr_err("%s: could not find display-timings node\n",
-			of_node_full_name(np));
-		return NULL;
+		timings_np = of_find_node_by_name(np, "display-timings");
+		if (!timings_np) {
+			pr_err("%s: could not find display-timings node\n",
+					of_node_full_name(np));
+			return NULL;
+		}
 	}
 
 	disp = kzalloc(sizeof(*disp), GFP_KERNEL);
@@ -259,9 +279,16 @@ int of_display_timings_exist(struct device_node *np)
 	if (!np)
 		return -EINVAL;
 
-	timings_np = of_parse_phandle(np, "display-timings", 0);
-	if (!timings_np)
-		return -EINVAL;
+	if (screen_alternate == 1)
+		timings_np = of_parse_phandle(np, "display-timings-alternate", 0);
+	else
+		timings_np = NULL;
+
+	if (!timings_np){
+		timings_np = of_parse_phandle(np, "display-timings", 0);
+		if (!timings_np)
+			return -EINVAL;
+	}
 
 	of_node_put(timings_np);
 	return 1;
-- 
2.1.4

