From 55e2f9ea2158f4bc90e50bc3602e6720105397ec Mon Sep 17 00:00:00 2001
From: Ruslan Filipovich <ruslan.filipovich@promwad.com>
Date: Sun, 22 May 2016 14:07:05 +0300
Subject: [PATCH 34/34] variscite: fix reboot command for variscite boards

---
 arch/arm/mach-imx/system.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/arch/arm/mach-imx/system.c b/arch/arm/mach-imx/system.c
index b73fabe..f75e98c 100644
--- a/arch/arm/mach-imx/system.c
+++ b/arch/arm/mach-imx/system.c
@@ -142,6 +142,7 @@ void mxc_restart(enum reboot_mode mode, const char *cmd)
 
 	if (cpu_is_mx1())
 		wcr_enable = (1 << 0);
+#if 0 /* variscite boards not used wcr_enable = 0x14 mode */
 	/*
 	 * Some i.MX6 boards use WDOG2 to reset external pmic in bypass mode,
 	 * so do WDOG2 reset here. Do not set SRS, since we will
@@ -154,6 +155,7 @@ void mxc_restart(enum reboot_mode mode, const char *cmd)
 			cpu_is_imx6sl())) || cpu_is_imx6sx() || cpu_is_imx7d()
 			|| cpu_is_imx6ul())
 		wcr_enable = 0x14;
+#endif
 	else
 		wcr_enable = (1 << 2);
 
-- 
2.1.4

