From 36b1236f38723c3734e73027811b99f97f6dd190 Mon Sep 17 00:00:00 2001
From: Harshesh Valera <harshesh.v@variscite.com>
Date: Mon, 14 Aug 2023 23:39:53 -0700
Subject: [PATCH] bluetooth: Update generic hw_settlement delay to 200 msec

Signed-off-by: Harshesh Valera <harshesh.v@variscite.com>
---
 src/hardware.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/hardware.c b/src/hardware.c
index 701cb86..368e671 100644
--- a/src/hardware.c
+++ b/src/hardware.c
@@ -234,7 +234,7 @@ static uint8_t bt_sco_i2spcm_param[SCO_I2SPCM_PARAM_SIZE] =
 static const fw_settlement_entry_t fw_settlement_table[] = {
     {"BCM43241", 200},
     {"BCM43341", 100},
-    {(const char *) NULL, 100}  // Giving the generic fw settlement delay setting.
+    {(const char *) NULL, 200}  // Giving the generic fw settlement delay setting.
 };
 
 
@@ -316,6 +316,7 @@ uint32_t look_up_fw_settlement_delay (void)
 
     BTHWDBG( "Settlement delay -- %d ms", ret_value);
 
+    ALOGE (" IMX8MP: Settlement delay -- %d ms", ret_value);
     return (ret_value);
 }
 
-- 
2.25.1

