From e6a549fe50c2c8fc192d193d23484f3d13adde9c Mon Sep 17 00:00:00 2001
From: mohit <mohit.p@variscite.com>
Date: Tue, 25 Jan 2022 09:35:43 -0800
Subject: [PATCH 3/4] camera: fix max resolution to 720p

---
 camera/CameraDeviceHWLImpl.cpp | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/camera/CameraDeviceHWLImpl.cpp b/camera/CameraDeviceHWLImpl.cpp
index 009961b..ba5c245 100644
--- a/camera/CameraDeviceHWLImpl.cpp
+++ b/camera/CameraDeviceHWLImpl.cpp
@@ -252,6 +252,9 @@ status_t CameraDeviceHwlImpl::initSensorStaticData()
     int previewCnt = 0, pictureCnt = 0;
     struct v4l2_frmsizeenum cam_frmsize;
     struct v4l2_frmivalenum vid_frmval;
+    char value[PROPERTY_VALUE_MAX] = {0};
+
+    property_get("ro.build.product", value, "0");
     while (ret == 0) {
         memset(TmpStr, 0, 20);
         memset(&cam_frmsize, 0, sizeof(struct v4l2_frmsizeenum));
@@ -298,6 +301,9 @@ status_t CameraDeviceHwlImpl::initSensorStaticData()
             mPreviewResolutions[previewCnt++] = cam_frmsize.discrete.width;
             mPreviewResolutions[previewCnt++] = cam_frmsize.discrete.height;
         }
+
+        if(cam_frmsize.discrete.height == 720 && (strncasecmp("som_mx8mn", value, strlen(value)) == 0))
+            break;
     }  // end while
 
     mPreviewResolutionCount = previewCnt;
-- 
2.25.1

