From 8c7144b3756c9f78ff0d8ee8d3b135d6f7a847de Mon Sep 17 00:00:00 2001
From: Harshesh Valera <harshesh.v@variscite.com>
Date: Sat, 7 Jan 2023 19:50:21 -0800
Subject: [PATCH 4/4] Fix Auto Focus Frame Freeze

Signed-off-by: Harshesh Valera <harshesh.v@variscite.com>
---
 camera/CameraDeviceHWLImpl.cpp | 12 ++++++------
 1 file changed, 6 insertions(+), 6 deletions(-)

diff --git a/camera/CameraDeviceHWLImpl.cpp b/camera/CameraDeviceHWLImpl.cpp
index ba5c245..c28687a 100644
--- a/camera/CameraDeviceHWLImpl.cpp
+++ b/camera/CameraDeviceHWLImpl.cpp
@@ -56,10 +56,6 @@ std::unique_ptr<CameraDeviceHwl> CameraDeviceHwlImpl::Create(
         return nullptr;
     }
 
-    ctrl_fd = open(*mDevPath[0], O_RDWR);
-    if (ctrl_fd < 0) {
-	ALOGE("%s invalid fd handle", __func__);
-    }
     ALOGI("%s: Created CameraDeviceHwlImpl for camera %u", __func__, device->camera_id_);
 
     return std::unique_ptr<CameraDeviceHwl>(device);
@@ -76,8 +72,6 @@ CameraDeviceHwlImpl::CameraDeviceHwlImpl(
         physical_device_map_(std::move(physical_devices)),
         mCallback(callback)
 {
-    if (ctrl_fd > 0)
-       close(ctrl_fd);
     mDevPath = devPaths;
     for (int i = 0; i<mDevPath.size(); ++i) {
         ALOGE("%s, mDevPath[%d] %s", __func__, i, *mDevPath[i]);
@@ -93,12 +87,18 @@ CameraDeviceHwlImpl::CameraDeviceHwlImpl(
 
     memset(mPreviewResolutions, 0, sizeof(mPreviewResolutions));
     memset(mPictureResolutions, 0, sizeof(mPictureResolutions));
+    ctrl_fd = open(*mDevPath[0], O_RDWR);
+    if (ctrl_fd < 0) {
+        ALOGE("%s invalid fd handle", __func__);
+    }
     memset(&caps_supports, 0, sizeof(caps_supports));
     m_raw_v4l2_format = -1;
 }
 
 CameraDeviceHwlImpl::~CameraDeviceHwlImpl()
 {
+    if (ctrl_fd > 0)
+       close(ctrl_fd);
     if(m_meta) {
         delete m_meta;
         m_meta = NULL;
-- 
2.25.1

