From fee5ae9cfb19babe08f6500001cc576f9503d38e Mon Sep 17 00:00:00 2001
From: varigigi <pierluigi.p@variscite.com>
Date: Fri, 7 Apr 2017 22:37:59 +0200
Subject: [PATCH] ota: don't copy uboot to ota package

---
 core/Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/core/Makefile b/core/Makefile
index 67b5d49..7ce58b8 100644
--- a/core/Makefile
+++ b/core/Makefile
@@ -1949,7 +1952,7 @@ ifdef BOARD_KERNEL_DTS
 endif
 ifdef DTS_PLATFORM
 	$(hide) mkdir -p $(zip_root)/RADIO;
-	$(hide) $(ACP) $(PRODUCT_OUT)/u-boot-$(DTS_PLATFORM).imx $(zip_root)/RADIO/bootloader.img
+#	$(hide) $(ACP) $(PRODUCT_OUT)/u-boot-$(DTS_PLATFORM).$(TARGET_BOOTLOADER_POSTFIX) $(zip_root)/RADIO/bootloader.img
 endif
 	$(hide) $(foreach t,$(INSTALLED_RADIOIMAGE_TARGET),\
 	            mkdir -p $(zip_root)/RADIO; \
-- 
2.7.4

