From 395a7a6f1667cb87576b6ac42ff7cde8a2c51c76 Mon Sep 17 00:00:00 2001
From: mohit <mohit.p@variscite.com>
Date: Tue, 2 Aug 2022 13:54:37 -0700
Subject: [PATCH 1/2] imx8mm_var_dart: Add dart_mx8mm target

---
 common/tools/imx-make.sh | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/common/tools/imx-make.sh b/common/tools/imx-make.sh
index 77010fa6..394ae225 100755
--- a/common/tools/imx-make.sh
+++ b/common/tools/imx-make.sh
@@ -194,7 +194,7 @@ if [ ${build_android_flag} -eq 1 ] || [ ${build_whole_android_flag} -eq 1 ]; the
     if [ -n "${build_bootimage}" ] || [ ${build_whole_android_flag} -eq 1 ]; then
         if [ ${TARGET_PRODUCT} = "evk_8mp" ] || [ ${TARGET_PRODUCT} = "evk_8mn" ] \
         || [ ${TARGET_PRODUCT} = "evk_8ulp" ] \
-        || [ ${TARGET_PRODUCT} = "evk_8mm" ] || [ ${TARGET_PRODUCT} = "evk_8mq" ]; then
+        || [ ${TARGET_PRODUCT} = "evk_8mm" ] || [ ${TARGET_PRODUCT} = "evk_8mq" ] || [ ${TARGET_PRODUCT} = "dart_mx8mm" ] ; then
             mv ${OUT}/boot.img ${OUT}/boot-imx.img
             # sign prebuilt gki boot.img
             make bootimage
-- 
2.36.1

