From 76471747d48862b0da8337bb3bf48b16882f3056 Mon Sep 17 00:00:00 2001
From: Harshesh Valera <harshesh.v@variscite.com>
Date: Wed, 23 Feb 2022 16:31:01 -0800
Subject: [PATCH] imx8mp_var_dart: Build VVCAM driver for dart_imx8mp platform

Signed-off-by: Harshesh Valera <harshesh.v@variscite.com>
---
 common/tools/imx-make.sh | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/common/tools/imx-make.sh b/common/tools/imx-make.sh
index dfc906d8..b1a50e00 100755
--- a/common/tools/imx-make.sh
+++ b/common/tools/imx-make.sh
@@ -126,7 +126,7 @@ if [ ${build_bootloader_kernel_flag} -eq 0 ] && [ ${build_android_flag} -eq 0 ];
 fi
 
 # vvcam.ko need build with kernel each time to make sure "insmod vvcam.ko" works
-if [ -n "${build_kernel}" ] && [ ${TARGET_PRODUCT} = "evk_8mp" ]; then
+if [ -n "${build_kernel}" ] && [ ${TARGET_PRODUCT} = "dart_mx8mp" ]; then
     build_vvcam="vvcam";
     build_kernel_module_flag=1;
 fi
-- 
2.25.1

