From 4ec4cdd2a4274a8a46ceedcd77be0aef3c3ce12f Mon Sep 17 00:00:00 2001
From: mohit <mohit.p@variscite.com>
Date: Tue, 27 Jul 2021 18:19:20 -0700
Subject: [PATCH] imx8qm_var_spear: Add SCFW compilation for som_mx8qm

---
 common/tools/imx-make.sh | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/common/tools/imx-make.sh b/common/tools/imx-make.sh
index 48f11af1..26077b0a 100755
--- a/common/tools/imx-make.sh
+++ b/common/tools/imx-make.sh
@@ -113,7 +113,7 @@ for arg in ${args[*]} ; do
     esac
 done
 
-if [ ${TARGET_PRODUCT} = "som_mx8qx" ]; then
+if [ ${TARGET_PRODUCT} = "som_mx8qx" ] || [ ${TARGET_PRODUCT} = "som_mx8qm" ]; then
 	source $ANDROID_BUILD_TOP/device/variscite/imx8q/som_mx8q/uboot-firmware/scfw_build.sh ${TARGET_PRODUCT}
 fi
 
-- 
2.25.1

