From cc62a758301ebc7819240c6d324fe06daa5d58df Mon Sep 17 00:00:00 2001
From: "roman.pletnev" <roman.pletnev@promwad.com>
Date: Fri, 20 May 2016 08:00:40 +0300
Subject: [PATCH 1/4] added build wl18xx compact

---
 core/Makefile | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/core/Makefile b/core/Makefile
index 0ceab9b..f7b025f 100644
--- a/core/Makefile
+++ b/core/Makefile
@@ -1047,6 +1047,10 @@ $(eval kernel_modules_dest_list += $(call append-path,$(PRODUCT_OUT),$(_dest))))
 ifneq ($(strip $(kernel_modules_src_list)),)
 
 kernelmodules: $(TARGET_PREBUILT_KERNEL) | $(ACP)
+ifeq ($(USES_TI_MAC80211),true)
+	$(MAKE) -C hardware/ti/wlan/mac80211/compat_wl18xx defconfig-wl18xx $(KERNEL_ENV) $(KMOD_ENV)
+	$(MAKE) -C hardware/ti/wlan/mac80211/compat_wl18xx $(KERNEL_ENV) $(KMOD_ENV)
+endif
 	$(MAKE) -C kernel_imx modules $(KERNEL_ENV)
 	$(hide) $(foreach cf, $(TARGET_KERNEL_MODULES),				\
 	$(eval _src := $(call word-colon,1,$(cf)))				\
-- 
2.1.4

